mergify_engine/engine/__init__.py

```diff
@@ -239,15 +239,21 @@ async def run(
             summary="You can accept them at https://dashboard.mergify.io/",
         )
 
-    if ctxt.pull["base"]["repo"]["private"] and not ctxt.subscription.has_feature(
-        subscription.Features.PRIVATE_REPOSITORY
-    ):
-        ctxt.log.info("mergify disabled: private repository")
-        return check_api.Result(
-            check_api.Conclusion.FAILURE,
-            title="Mergify is disabled",
-            summary=ctxt.subscription.reason,
-        )
+    if ctxt.pull["base"]["repo"]["private"]:
+        if not ctxt.subscription.has_feature(subscription.Features.PRIVATE_REPOSITORY):
+            ctxt.log.info("mergify disabled: private repository")
+            return check_api.Result(
+                check_api.Conclusion.FAILURE,
+                title="Mergify is disabled",
+                summary=ctxt.subscription.reason,
+            )
+        elif not ctxt.subscription.has_feature(subscription.Features.PUBLIC_REPOSITORY):
```

Comment:
This is only going to be checked if the repo is private, not public :)

We want this check to be done if the repo is public.